File: src/bingda_tutorials/script/tutorials_tf_transform.py
#!/usr/bin/env python
import rospy
import tf
import math

def tf_transform():
    rospy.init_node('tutorials_tf_transform', anonymous=False)
    tf_broadcaster = tf.TransformBroadcaster()
    rate = rospy.Rate(10) # 10hz
    angle = 0.0
    rospy.loginfo('Start TF Transform')
    while not rospy.is_shutdown():
        current_time = rospy.Time.now()
        x = math.cos(angle)*0.3
        y = math.sin(angle)*0.3
        z = 0.2
        quat = tf.transformations.quaternion_from_euler(0,0,angle)
        tf_broadcaster.sendTransform((x,y,z),quat,current_time,'tf3','tf1')
        tf_broadcaster.sendTransform((x,y,z*2),(0,0,0,1),current_time,'tf4','tf1')
        angle += 0.01
        rate.sleep()

if __name__ == '__main__':
    tf_transform()

File: src/bingda_tutorials/src/tutorials_tf_transform.cpp
#include <ros/ros.h>
#include <tf/transform_broadcaster.h>
#include <tf/transform_datatypes.h>


int main(int argc, char** argv){
    ros::init(argc, argv, "robot_tf_publisher");
    ros::NodeHandle n;
    ros::Rate r(10);
    tf::TransformBroadcaster broadcaster;
    float angle = 0.0;
    float x = 0.0;
    float y = 0.0;
    float z = 0.0;
    ROS_INFO("Start TF Transform");
    while(n.ok()){

        x = cos(angle)*0.3;
        y = sin(angle)*0.3;
        z = 0.2;
        broadcaster.sendTransform(
            tf::StampedTransform(
                tf::Transform(tf::createQuaternionFromRPY(0.0,0.0,angle), tf::Vector3(x, y, z)),
                ros::Time::now(),"tf1", "tf3"));

        broadcaster.sendTransform(
            tf::StampedTransform(
                tf::Transform(tf::Quaternion(0, 0, 0, 1), tf::Vector3(x, y, z*2)),
                ros::Time::now(),"tf1", "tf4"));

        r.sleep();
        angle += 0.01;
    }
}